### DISCUSSION TITLE
Support class @Decorators

### DISCUSSION THREAD

**Discussion by georgekrax** [swc-project/swc#3032] [Q&A] [answered]
Hello everyone,

Does **swc** support `class` @Decorators? I have a Node.js project with **TypeORM**, and would to use swc for faster compilation. Is swc a good solution for a Node.js - Express.js project, or does it support only front-end projects, like Next.js?

Thanks in advance

**Comment by aelbore** [comment, marked as answer]
@georgekrax yes it support decorators you can try here in swc playground https://swc.rs/playground

you can try this config
```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "decorators": true,
      "dynamicImport": true
    },
    "target": "es2020"
  }
}
```

**Reply by georgekrax** [reply]
Oh, yes. I tried it today with the configuration you provided and worked wonderfully. Also, I am using the `paths` option in `tsconfig.json`, so the relative import paths should be somehow mapped to their actual paths in the directory. What can I use to make this work? Should I use the famous `tsconfig-paths` package, or is there a functionality that **SWC** provides?

**Reply by kwonoj** [reply]
Please refer doc https://swc.rs/docs/configuration/compilation#jscpaths and give it a try.

**Reply by georgekrax** [reply]
Oh, yes you are right. Thanks!

**Reply by georgekrax** [reply]
But still I cannot run it with `node build/index.js`. When I use `tsc` it works fine, but I do not why **swc** and **tsc** compile differently my code.

**Reply by kdy1** [reply]
Decorator is supported, but there are some bugs.

**Reply by georgekrax** [reply]
In my code it seems like when I run `node build/index.js`, it starts the server once and then restarts again without making any changes. I do not know if `nodemon` also creates this issue

**Reply by georgekrax** [reply]
Okay, I fixed it! However, `node build/index.js` takes 10+ seconds to run!!! Any solution?
